presto-orc/src/main/java/com/facebook/presto/orc/metadata/DwrfMetadataReader.java

```diff
@@ -105,7 +109,41 @@ public Footer readFooter(HiveWriterVersion hiveWriterVersion, InputStream inputS
                 toStripeInformation(footer.getStripesList()),
                 toType(footer.getTypesList()),
                 fileStats,
-                toUserMetadata(footer.getMetadataList()));
+                toUserMetadata(footer.getMetadataList()),
+                toEncryption(footer.getEncryption()));
+    }
+
+    private Optional<Encryption> toEncryption(DwrfProto.Encryption encryption)
+    {
+        if (encryption.getEncryptionGroupsCount() == 0) {
+            return Optional.empty();
+        }
+        KeyProvider keyProvider = toKeyProvider(encryption.getKeyProvider());
+        List<EncryptionGroup> encryptionGroups = toEncryptionGroups(encryption.getEncryptionGroupsList());
+        return Optional.of(new Encryption(keyProvider, encryptionGroups));
+    }
+
+    private List<EncryptionGroup> toEncryptionGroups(List<DwrfProto.EncryptionGroup> encryptionGroups)
+    {
+        ImmutableList.Builder<EncryptionGroup> encryptionGroupBuilder = ImmutableList.builder();
+        for (DwrfProto.EncryptionGroup dwrfEncryptionGroup : encryptionGroups) {
+            encryptionGroupBuilder.add(new EncryptionGroup(
+                    dwrfEncryptionGroup.getNodesList(),
+                    dwrfEncryptionGroup.hasKeyMetadata() ? Optional.of(byteStringToSlice(dwrfEncryptionGroup.getKeyMetadata())) : Optional.empty(),
+                    byteStringToSlice(dwrfEncryptionGroup.getStatistics())));
+        }
+        return encryptionGroupBuilder.build();
+    }
+
+    private KeyProvider toKeyProvider(DwrfProto.Encryption.KeyProvider keyProvider)
+    {
+        switch (keyProvider) {
+            case CRYPTO_SERVICE:
+                return KeyProvider.CRYPTO_SERVICE;
+            default:
+                // TODO: should we throw an exception instead?
+                return KeyProvider.UNKNOWN;
```

Comment:
@zzhao0  what's the correct behavior for UKNOWN key provider type?

Reply:
UNKNOWN is basically a stub for anything. For example, during test, we write a file using some fake key provider, and set it to unknown. As long as the reader uses the same key provider, it is able to read the file without making the fake provider an actual provider in the proto